common/playlist.c

```diff
@@ -214,6 +214,55 @@ struct playlist_entry *playlist_entry_get_rel(struct playlist_entry *e,
     return playlist_entry_from_index(e->pl, e->pl_index + direction);
 }
 
+struct playlist_entry *playlist_get_first_in_next_playlist(struct playlist *pl,
+                                                           int direction) {
+    struct playlist_entry *entry = playlist_get_next(pl, direction);
+    if (!entry)
+        return NULL;
+
+    while (entry && entry->playlist_path &&
+           strcmp(entry->playlist_path, pl->current->playlist_path) == 0)
+        entry = playlist_entry_get_rel(entry, direction);
+
+    if (direction < 0)
+        entry = playlist_get_first_in_same_playlist(entry,
+                                                    pl->current->playlist_path);
+
+    return entry;
+}
+
+struct playlist_entry *playlist_get_first_in_same_playlist(
+    struct playlist_entry *entry, char *current_playlist_path) {
+    void *tmp = talloc_new(NULL);
+
+    if (!entry || !entry->playlist_path)
+        goto exit;
+
+    char* playlist_path = talloc_strdup(tmp, entry->playlist_path);
+    mp_path_strip_trailing_separator(playlist_path);
+    if (bstr_startswith(bstr0(current_playlist_path),
+                        bstr0(talloc_strdup_append(playlist_path, "/")))
+#if HAVE_DOS_PATHS
+        ||
+        bstr_startswith(bstr0(current_playlist_path),
+                        bstr0(talloc_strdup_append(playlist_path, "\\")))
+#endif
```

Comment:
Why strip the separator if you're going to compare them again with the separator? Couldn't you just do that after this check or am I missing something?

Reply:
We don't want to trigger to condition on e.g. `foobar/` starting with `foo/`.

Reply:
Added a comment explaining the issue in detail.

Reply:
Ah okay I was wondering why you couldn't just strip all separators but that makes sense now. I might be nicer to add a `mp_path_add_trailing_separator` in path.c so we can keep the DOS_PATHS thing out of here.

Reply:
Windows is weird and evidently you have to check both so nevermind.